x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -938,6 +1055,39 @@ private static Map<String, Object> maybeRewriteMetadataForApiKeyRoleDescriptors(
         return metadata;
     }
 
+    // pkg-private for testing
+    static Map<String, Object> maybeRewriteMetadataForRemoteAccessAuthentication(
+        final TransportVersion olderVersion,
+        final Authentication authentication
+    ) {
+        // TODO also handle remote access authentication executed with old API keys (either by rejecting this, or rewriting role
+        // descriptors)
```

Comment:
It's a weird edge case but technically nothing stands in the way of an admin taking an old API key that has something like `superuser` permissions and handing that to a querying cluster to use as a remote access key. I think it's fair for us to prevent this with a version check on the API key during authentication but it's worth failing gracefully

Reply:
I may be missing something. Why is an old API key from FC with `superuser` permission a problem when it is used as a remote access key? It has more privileges than necessary but that is not an unique problem for old API keys. The creation version of the API key is *not* the same as the authentication's version. The authentication's version should always be `CURRENT` from a successful authentication process. As such, the role descriptors should be stored as bytes even if the API key itself is old.

Reply:
You're right, I had my wires crossed: to paraphrase, since we only introduced remote access in 8.7, by the time we get to the rewrite method, (remote access) authentication based on an old (map-based role descriptors) API key will always already have the role descriptors written in its metadata as bytes, and we don't need to worry about having to rewrite them here. 

I will remove the nonsensical TODO 👍